include/bluetooth/hci.h

```diff
@@ -2873,6 +2873,22 @@ typedef bool bt_hci_vnd_evt_cb_t(struct net_buf_simple *buf);
   */
 int bt_hci_register_vnd_evt_cb(bt_hci_vnd_evt_cb_t cb);
 
+/** @brief Get Random bytes from the LE Controller.
+ *
+ * Send the HCI_LE_Rand to the LE Controller as many times as required to
+ * fill the provided @p buffer.
+ *
+ * @note This function is provided as a helper to gather an arbitrary number of
+ * random bytes from an LE Controller using the HCI_LE_Rand command.
+ *
+ * @param buffer Buffer to fill with random bytes.
+ * @param len Length of the buffer in bytes.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int bt_hci_le_rand(uint8_t *buffer, uint16_t len);
```

Comment:
To maximise usability and to avoid unnecessary explicit casts for thee caller, I'd recommend the input parameter types to be `(void *buf, size_t len)`

Reply:
Indeed you are right.